core/src/main/scala/kafka/zk/AdminZkClient.scala

```diff
@@ -221,15 +201,15 @@ class AdminZkClient(zkClient: KafkaZkClient) extends Logging {
       AdminUtils.assignReplicasToBrokers(allBrokers, partitionsToAdd, existingAssignmentPartition0.size,
         startIndex, existingAssignment.size)
     }
+
     val proposedAssignment = existingAssignment ++ proposedAssignmentForNewPartitions
     if (!validateOnly) {
       info(s"Creating $partitionsToAdd partitions for '$topic' with the following replica assignment: " +
         s"$proposedAssignmentForNewPartitions.")
-      // add the combined new list
-      createOrUpdateTopicPartitionAssignmentPathInZK(topic, proposedAssignment, update = true)
+
+      writeTopicPartitionAssignment(topic, proposedAssignment, isUpdate = true)
     }
     proposedAssignment
-
   }
 
   private def validateReplicaAssignment(replicaAssignment: Map[Int, Seq[Int]],
```

Comment:
Shouldn't there be a check to validate if all replicas passed are existing and available brokers during topic creation? 
PR #5665 has a check to guarantee at least one replica is alive during topic creation. Just take a look. 
Also, shouldn't we refactor and maintain uniformity in [ReassignPartitionsCommand](https://github.com/apache/kafka/blob/928f45f61f793f0b2d7df11ff92463d7edda60d6/core/src/main/scala/kafka/admin/ReassignPartitionsCommand.scala#L316) validations and [AdminZkClient](https://github.com/apache/kafka/blob/928f45f61f793f0b2d7df11ff92463d7edda60d6/core/src/main/scala/kafka/zk/AdminZkClient.scala#L120) validations?